src/main.ts

```diff
@@ -119,11 +119,11 @@ async function main(): Promise<void> {
     await checkMarketplaceAddress(MarketplaceJSON.abi, marketplaceContract)
 
     await watcher.on("subscribed", async (args: {
-            blockNumber: number,
-            blockIndex: number,
-            product: string,
-            address: EthereumAddress,
-            endsAt: string,
```

Comment:
Just a note that the API expects `endsAt` to be a number, not a string - but if this same code worked before, then I assume Grails does some automagic conversion there and it will work.